docs/ManualMocks.md

```diff
@@ -181,3 +181,36 @@ the test file. But often you need to instruct Jest to use a mock before modules
 use it. For this reason, Jest will automatically hoist `jest.mock` calls to the
 top of the module (before any imports). To learn more about this and see it in
 action, see [this repo](https://github.com/kentcdodds/how-jest-mocking-works).
+
+### Mocking methods which are not implemented in JSDOM
+If some code uses a method which JSDOM (the DOM implementation used by Jest)
+hasn't implemented yet, testing it is not easily possible. This is e.g. the
+case with `window.matchMedia()`. Jest returns `TypeError: window.matchMedia is 
+not a function` and doesn't properly execute the test. 
+
+In this case, a manual mock in the test file should solve the issue:
+```js
+window.matchMedia = jest.fn().mockImplementation(query => {
+  return {
+    matches: false,
+    media: query,
+    onchange: null,
+    addListener: jest.fn(),
+    removeListener: jest.fn(),
+  };
+});
+```
+
+This works if `window.matchMedia()` is used in a function (or method) which is
+invoked in the test. If `window.matchMedia()` is executed directly in the
+tested file, Jest reports the same error. In this case, the solution is to move
+the manual mock into a separate file and include this one in the test
+**before** the tested file:
+```js
+import './matchMedia.mock'; // Must be imported before the tested file
+import { myMethod } from './file-to-test';
+
+describe('myMethod()', () => {
+    ...
+});
```

Comment:
I'm not sure I understand this second example, can you expand? Would this be better solved in other ways?

Reply:
Sure (the code is rather stupid but should illustrate the issue):
```js
window.matchMedia("(min-width: 800px)") // --1--

export function matchMediaQuery() {
  return window.matchMedia("(min-width: 800px)").matches // --2--
}
```

`window.matchMedia()` in --2-- is executed when the test is run. In this case, mocking `matchMedia` inside the test file works.

`window.matchMedia()` in --1-- requires to import the monkey patched `matchMedia` before importing the tested file. Otherwise the test fails.